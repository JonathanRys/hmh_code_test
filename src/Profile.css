.profile {
  display: inline-block;
  padding: 10px;
  text-align: right;
}

.profile h3 {
  text-align: center;
}

.profile input {
  margin: 3px 5px;
}
