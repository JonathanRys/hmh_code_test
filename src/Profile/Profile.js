import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Profile.css'

class Profile extends Component {
    constructor(props) {
        super(props);

        this.state = {
            data: {
                firstName: "",
                lastName: "",
                startDate: ""
            }
        }
    }

    onBlur = (e) => {
        this.setState({
            data: {
                ...this.state.data,
                [e.target.name]: e.target.value
            }
        })
    }

    onClick = () => {
        this.props.onClick({ data: this.state.data })
    }

    render() {
        return (
            <div className="profile">
                <h3>Profile</h3>
                <label>First Name:</label><input name="firstName" type="text" onBlur={this.onBlur} /><br />
                <label>Last Name:</label><input name="lastName" type="text" onBlur={this.onBlur} /><br />
                <label>Start Date:</label><input name="startDate" type="text" onBlur={this.onBlur} /><br />
                <input type="button" value="Save" onClick={this.onClick} />
            </div>
        )
    }
}

Profile.propTypes = {
    store: PropTypes.object.isRequired
}

export default Profile
