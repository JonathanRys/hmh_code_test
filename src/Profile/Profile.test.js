import React from 'react';
import ReactDOM from 'react-dom';
import Profile from './Profile';
import { render, shallow } from 'enzyme'

import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter })

const mockData = {
    data: {
        firstName: "Lester",
        lastName: "Tester",
        favColor: "Beige"
    }
}

const mockOnClick = data => {
    expect(data).toEqual(mockData)
}

describe('Test Profile component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Profile />, div);
    });

    it('passes a snapshot test', () => {
        const profile = render(<Profile onClick={mockOnClick} />)
        expect(profile).toMatchSnapshot()
    });

    it('saves data from inputs', () => {
        const profile = shallow(<Profile onClick={mockOnClick} />)
        const inputs = profile.find("input")
        const button = inputs.find('.save-button')

        const firstNameInput = inputs.at(0)
        const lastNameInput = inputs.at(1)
        const favColorInput = inputs.at(2)

        firstNameInput.simulate("blur", { target: { name: "firstName", value: "Lester" } })
        lastNameInput.simulate("blur", { target: { name: "lastName", value: "Tester" } })
        favColorInput.simulate("blur", { target: { name: "favColor", value: "Beige" } })

        button.simulate("click")
    })
});
