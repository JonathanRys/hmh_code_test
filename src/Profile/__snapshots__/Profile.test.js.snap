// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile component renders without crashing 1`] = `
Profile {
  "_reactInternalFiber": FiberNode {
    "_debugID": 18,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": null,
    "alternate": null,
    "child": FiberNode {
      "_debugID": 19,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": null,
      "alternate": null,
      "child": FiberNode {
        "_debugID": 20,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": null,
        "alternate": null,
        "child": null,
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": "Profile",
        },
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": [Circular],
        "sibling": FiberNode {
          "_debugID": 21,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": null,
          "alternate": null,
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "internalContextTag": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": "First Name:",
          },
          "memoizedState": null,
          "nextEffect": null,
          "pendingProps": null,
          "ref": null,
          "return": [Circular],
          "sibling": FiberNode {
            "_debugID": 22,
            "_debugIsCurrentlyTiming": false,
            "_debugOwner": [Circular],
            "_debugSource": null,
            "alternate": null,
            "child": null,
            "effectTag": 0,
            "expirationTime": 0,
            "firstEffect": null,
            "index": 2,
            "internalContextTag": 0,
            "key": null,
            "lastEffect": null,
            "memoizedProps": Object {
              "name": "firstName",
              "onBlur": [Function],
              "type": "text",
            },
            "memoizedState": null,
            "nextEffect": null,
            "pendingProps": null,
            "ref": null,
            "return": [Circular],
            "sibling": FiberNode {
              "_debugID": 23,
              "_debugIsCurrentlyTiming": false,
              "_debugOwner": [Circular],
              "_debugSource": null,
              "alternate": null,
              "child": null,
              "effectTag": 0,
              "expirationTime": 0,
              "firstEffect": null,
              "index": 3,
              "internalContextTag": 0,
              "key": null,
              "lastEffect": null,
              "memoizedProps": Object {},
              "memoizedState": null,
              "nextEffect": null,
              "pendingProps": null,
              "ref": null,
              "return": [Circular],
              "sibling": FiberNode {
                "_debugID": 24,
                "_debugIsCurrentlyTiming": false,
                "_debugOwner": [Circular],
                "_debugSource": null,
                "alternate": null,
                "child": null,
                "effectTag": 0,
                "expirationTime": 0,
                "firstEffect": null,
                "index": 4,
                "internalContextTag": 0,
                "key": null,
                "lastEffect": null,
                "memoizedProps": Object {
                  "children": "Last Name:",
                },
                "memoizedState": null,
                "nextEffect": null,
                "pendingProps": null,
                "ref": null,
                "return": [Circular],
                "sibling": FiberNode {
                  "_debugID": 25,
                  "_debugIsCurrentlyTiming": false,
                  "_debugOwner": [Circular],
                  "_debugSource": null,
                  "alternate": null,
                  "child": null,
                  "effectTag": 0,
                  "expirationTime": 0,
                  "firstEffect": null,
                  "index": 5,
                  "internalContextTag": 0,
                  "key": null,
                  "lastEffect": null,
                  "memoizedProps": Object {
                    "name": "lastName",
                    "onBlur": [Function],
                    "type": "text",
                  },
                  "memoizedState": null,
                  "nextEffect": null,
                  "pendingProps": null,
                  "ref": null,
                  "return": [Circular],
                  "sibling": FiberNode {
                    "_debugID": 26,
                    "_debugIsCurrentlyTiming": false,
                    "_debugOwner": [Circular],
                    "_debugSource": null,
                    "alternate": null,
                    "child": null,
                    "effectTag": 0,
                    "expirationTime": 0,
                    "firstEffect": null,
                    "index": 6,
                    "internalContextTag": 0,
                    "key": null,
                    "lastEffect": null,
                    "memoizedProps": Object {},
                    "memoizedState": null,
                    "nextEffect": null,
                    "pendingProps": null,
                    "ref": null,
                    "return": [Circular],
                    "sibling": FiberNode {
                      "_debugID": 27,
                      "_debugIsCurrentlyTiming": false,
                      "_debugOwner": [Circular],
                      "_debugSource": null,
                      "alternate": null,
                      "child": null,
                      "effectTag": 0,
                      "expirationTime": 0,
                      "firstEffect": null,
                      "index": 7,
                      "internalContextTag": 0,
                      "key": null,
                      "lastEffect": null,
                      "memoizedProps": Object {
                        "children": "Favorite Color:",
                      },
                      "memoizedState": null,
                      "nextEffect": null,
                      "pendingProps": null,
                      "ref": null,
                      "return": [Circular],
                      "sibling": FiberNode {
                        "_debugID": 28,
                        "_debugIsCurrentlyTiming": false,
                        "_debugOwner": [Circular],
                        "_debugSource": null,
                        "alternate": null,
                        "child": null,
                        "effectTag": 0,
                        "expirationTime": 0,
                        "firstEffect": null,
                        "index": 8,
                        "internalContextTag": 0,
                        "key": null,
                        "lastEffect": null,
                        "memoizedProps": Object {
                          "name": "favColor",
                          "onBlur": [Function],
                          "type": "text",
                        },
                        "memoizedState": null,
                        "nextEffect": null,
                        "pendingProps": null,
                        "ref": null,
                        "return": [Circular],
                        "sibling": FiberNode {
                          "_debugID": 29,
                          "_debugIsCurrentlyTiming": false,
                          "_debugOwner": [Circular],
                          "_debugSource": null,
                          "alternate": null,
                          "child": null,
                          "effectTag": 0,
                          "expirationTime": 0,
                          "firstEffect": null,
                          "index": 9,
                          "internalContextTag": 0,
                          "key": null,
                          "lastEffect": null,
                          "memoizedProps": Object {},
                          "memoizedState": null,
                          "nextEffect": null,
                          "pendingProps": null,
                          "ref": null,
                          "return": [Circular],
                          "sibling": FiberNode {
                            "_debugID": 30,
                            "_debugIsCurrentlyTiming": false,
                            "_debugOwner": [Circular],
                            "_debugSource": null,
                            "alternate": null,
                            "child": null,
                            "effectTag": 0,
                            "expirationTime": 0,
                            "firstEffect": null,
                            "index": 10,
                            "internalContextTag": 0,
                            "key": null,
                            "lastEffect": null,
                            "memoizedProps": Object {
                              "onClick": [Function],
                              "type": "button",
                              "value": "Save",
                            },
                            "memoizedState": null,
                            "nextEffect": null,
                            "pendingProps": null,
                            "ref": null,
                            "return": [Circular],
                            "sibling": null,
                            "stateNode": <input
                              type="button"
                              value="Save"
                            />,
                            "tag": 5,
                            "type": "input",
                            "updateQueue": null,
                          },
                          "stateNode": <br />,
                          "tag": 5,
                          "type": "br",
                          "updateQueue": null,
                        },
                        "stateNode": <input
                          name="favColor"
                          type="text"
                        />,
                        "tag": 5,
                        "type": "input",
                        "updateQueue": null,
                      },
                      "stateNode": <label>
                        Favorite Color:
                      </label>,
                      "tag": 5,
                      "type": "label",
                      "updateQueue": null,
                    },
                    "stateNode": <br />,
                    "tag": 5,
                    "type": "br",
                    "updateQueue": null,
                  },
                  "stateNode": <input
                    name="lastName"
                    type="text"
                  />,
                  "tag": 5,
                  "type": "input",
                  "updateQueue": null,
                },
                "stateNode": <label>
                  Last Name:
                </label>,
                "tag": 5,
                "type": "label",
                "updateQueue": null,
              },
              "stateNode": <br />,
              "tag": 5,
              "type": "br",
              "updateQueue": null,
            },
            "stateNode": <input
              name="firstName"
              type="text"
            />,
            "tag": 5,
            "type": "input",
            "updateQueue": null,
          },
          "stateNode": <label>
            First Name:
          </label>,
          "tag": 5,
          "type": "label",
          "updateQueue": null,
        },
        "stateNode": <h3>
          Profile
        </h3>,
        "tag": 5,
        "type": "h3",
        "updateQueue": null,
      },
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          <h3>
            Profile
          </h3>,
          <label>
            First Name:
          </label>,
          <input
            name="firstName"
            onBlur={[Function]}
            type="text"
          />,
          <br />,
          <label>
            Last Name:
          </label>,
          <input
            name="lastName"
            onBlur={[Function]}
            type="text"
          />,
          <br />,
          <label>
            Favorite Color:
          </label>,
          <input
            name="favColor"
            onBlur={[Function]}
            type="text"
          />,
          <br />,
          <input
            onClick={[Function]}
            type="button"
            value="Save"
          />,
        ],
        "className": "profile",
      },
      "memoizedState": null,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <div
        class="profile"
      >
        <h3>
          Profile
        </h3>
        <label>
          First Name:
        </label>
        <input
          name="firstName"
          type="text"
        />
        <br />
        <label>
          Last Name:
        </label>
        <input
          name="lastName"
          type="text"
        />
        <br />
        <label>
          Favorite Color:
        </label>
        <input
          name="favColor"
          type="text"
        />
        <br />
        <input
          type="button"
          value="Save"
        />
      </div>,
      "tag": 5,
      "type": "div",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "internalContextTag": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {
      "onClick": [Function],
    },
    "memoizedState": Object {
      "data": Object {
        "favColor": "",
        "firstName": "",
        "lastName": "",
      },
    },
    "nextEffect": null,
    "pendingProps": null,
    "ref": null,
    "return": FiberNode {
      "_debugID": 16,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 16,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "internalContextTag": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <div
              class="profile"
            >
              <h3>
                Profile
              </h3>
              <label>
                First Name:
              </label>
              <input
                name="firstName"
                type="text"
              />
              <br />
              <label>
                Last Name:
              </label>
              <input
                name="lastName"
                type="text"
              />
              <br />
              <label>
                Favorite Color:
              </label>
              <input
                name="favColor"
                type="text"
              />
              <br />
              <input
                type="button"
                value="Save"
              />
            </div>
          </div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "hydrate": false,
          "isReadyForCommit": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Profile
                onClick={[Function]}
              />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "nextCallback": null,
            "partialState": Object {
              "element": <Profile
                onClick={[Function]}
              />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "internalContextTag": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Profile
          onClick={[Function]}
        />,
      },
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <div
            class="profile"
          >
            <h3>
              Profile
            </h3>
            <label>
              First Name:
            </label>
            <input
              name="firstName"
              type="text"
            />
            <br />
            <label>
              Last Name:
            </label>
            <input
              name="lastName"
              type="text"
            />
            <br />
            <label>
              Favorite Color:
            </label>
            <input
              name="favColor"
              type="text"
            />
            <br />
            <input
              type="button"
              value="Save"
            />
          </div>
        </div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "hydrate": false,
        "isReadyForCommit": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": null,
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "onBlur": [Function],
  "onClick": [Function],
  "props": Object {
    "onClick": [Function],
  },
  "refs": Object {},
  "state": Object {
    "data": Object {
      "favColor": "",
      "firstName": "",
      "lastName": "",
    },
  },
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
